src/platform/datapath_raw_socket.c

```diff
@@ -499,7 +499,13 @@ typedef struct ETHERNET_HEADER {
 
 typedef struct IPV4_HEADER {
     uint8_t VersionAndHeaderLength;
-    uint8_t TypeOfServiceAndEcnField;
+    union {
+        uint8_t TypeOfServiceAndEcnField;
+        struct {
```

Comment:
Could you complete the `struct` part, so we're not missing 4 bits?

Reply:
Wow yeah, I think this was a typo- TypeOfService should be 6 bits. It just happened to still work with the typo